File: src/main/java/com/dik/models/Operation.java
package com.dik.models;

/**
 * Created by billa on 2016-12-04.
 */
public class Operation implements Comparable<Operation> {
    private String name;
    private float time;

    public Operation(String name, float time) {
        this.name = name;
        this.time = time;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public float getTime() {
        return time;
    }

    public void setTime(float time) {
        this.time = time;
    }

    @Override
    public int compareTo(Operation o) {
        return name.compareTo(o.getName());
    }
}

File: app/models/Operation.scala
package models

import dbgeneratedtable.Tables


/**
  * Created by billa on 2016-12-15.
  */

trait Operation{
  def getId():Int
  def getName():String
}

object AssemblyOperation{
  def mapAssemblyOperationRowToModel(rowObj:Tables.AssemblyOperationMappingRow,operationObj:Operation): AssemblyOperation ={
      new AssemblyOperation(rowObj.operationId,operationObj.getName(),rowObj.operationTime)
  }
}

class AssemblyOperation(private val id:Int =0,private val  name:String,private val time:Float) extends Operation(){
  def getOperationTime():Float =time

  override def getId(): Int = id

  override def getName(): String = name

  /**
    * Check test cases for all condition for equals i.e. consistensy,transivity,reflexive
    * @param obj
    */
  override def equals(obj: scala.Any): Boolean = {
    obj match {
      case null => false
      case obj:AssemblyOperation => getName().equalsIgnoreCase(obj.getName()) &&
        getId().compareTo(obj.getId())==0 &&
        getOperationTime().compareTo(obj.getOperationTime()) ==0
      case _ => false
    }
  }
}


class ComponentOperation(private val id:Int =0,private val  name:String) extends Operation(){


  /**
    * Check test cases for all condition for equals i.e. consistensy,transivity,reflexive
    * @param obj
    */
  override def equals(obj: scala.Any): Boolean = {
    obj match {
      case null => false
      case obj:Operation => getName().equalsIgnoreCase(obj.getName()) && getId().compareTo(obj.getId())==0
      case _ => false
    }
  }

  override def getId(): Int = id

  override def getName(): String = name

}
